tapgarden/planter.go

```diff
@@ -2203,6 +2203,69 @@ func matchPsbtToGroupReq(psbt psbt.Packet,
 	return fn.None[asset.GroupKeyRequest](), nil
 }
 
+// sealBatchPreCommit injects the group public key obtained during the sealing
+// phase into the pre‑commitment output descriptor of the batch's genesis
+// packet.
+//
+// Preconditions:
+//   - batch.UniverseCommitments must be true – otherwise the function is a NOP.
+//   - batch.GenesisPacket must not be nil.
+//
+// Post‑conditions:
+//   - batch.GenesisPacket.PreCommitmentOutput is populated with the group key.
+//
+// NOTE: The function mutates the supplied *MintingBatch in place.
+func sealBatchPreCommit(batch *MintingBatch) error {
+	// Fast‑exit if Universe Commitments are disabled – nothing to update.
+	if !batch.UniverseCommitments {
+		return nil
+	}
+
+	// A valid genesis packet is mandatory once Universe Commitments are on.
+	if batch.GenesisPacket == nil {
+		return fmt.Errorf("batch genesis packet is unexpectedly " +
+			"nil, cannot update mint anchor pre-commitment output")
+	}
+
+	// Retrieve the group public key recorded during sealing.
+	groupKeyOpt, err := fetchPreCommitGroupKey(batch)
+	if err != nil {
+		return fmt.Errorf("unable to fetch pre-commit group key: %w",
+			err)
+	}
+
+	groupKey, err := groupKeyOpt.UnwrapOrErr(
+		fmt.Errorf("pre-commitment output group key is unexpectedly " +
+			"absent"),
+	)
+	if err != nil {
+		return err
+	}
+
+	// Ensure that the group key is set in the genesis packet
+	// pre-commitment output descriptor.
+	fundedAnchor := batch.GenesisPacket
+	if fundedAnchor == nil {
+		return fmt.Errorf("funded anchor is unexpectedly nil, " +
+			"cannot update mint anchor pre-commitment output " +
+			"descriptor")
+	}
+
+	preCommitDesc, err :=
+		fundedAnchor.PreCommitmentOutput.UnwrapOrErr(
+			fmt.Errorf("pre-commitment output is unexpectedly " +
+				"absent"),
+		)
+	if err != nil {
+		return err
+	}
+
+	preCommitDesc.GroupPubKey = fn.Some(groupKey)
```

Comment:
Why do we unpack this when we put it back as an option? If we just want to validate it's there, we can do `if !groupKeyOpt.IsSome()`. Same for the preCommitDesc. Probably doesn't matter too much, but might visually simplify the logic a little bit.